07.ls_object/lib/ls_command.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require 'optparse'
+require 'pathname'
+require_relative 'ls_file'
+
+class LsCommand
+  COLS = 3
+
+  def show_files
+    options = parse_options
+    ls_files = build_ls_files(options)
+    if options[:long]
+      show_files_in_long_format(ls_files)
+    else
+      show_files_in_default_format(ls_files)
+    end
+  end
+
+  private
+
+  def parse_options
+    options = { all: false, reverse: false, long: false }
+
+    OptionParser.new do |opts|
+      opts.on('-a') { options[:all] = true }
+      opts.on('-r') { options[:reverse] = true }
+      opts.on('-l') { options[:long] = true }
+    end.parse!
+
+    options
+  end
+
+  def build_ls_files(options)
+    target_directory = Pathname(ARGV[0] || '.')
+    files = Dir.foreach(target_directory)
+    ls_files = files.map { |file| LsFile.new(file, target_directory, options) }
+    filtered_ls_files = options[:all] ? ls_files : ls_files.reject(&:hidden?)
+    options[:reverse] ? filtered_ls_files.sort.reverse : filtered_ls_files.sort
+  end
+
+  def show_files_in_long_format(ls_files)
+    puts "total #{ls_files.sum(&:block_size)}"
+    max_length = find_file_max_length(ls_files)
+    ls_files.each do |ls_file|
+      ls_file_properties = []
+      ls_file_properties << ls_file.mode
+      ls_file_properties << ls_file.hard_links.to_s.rjust(max_length[:hard_links])
+      ls_file_properties << ls_file.owner_name.to_s.ljust(max_length[:owner_name])
+      ls_file_properties << ls_file.owner_group_name.to_s.ljust(max_length[:owner_group_name])
+      ls_file_properties << ls_file.size_or_device_info.to_s.rjust(max_length[:size])
+      ls_file_properties << ls_file.last_modified_time.strftime(make_time_format(ls_file))
```

Comment:
```suggestion
      ls_file_properties << format_last_modified_time(ls_file)
```

make_time_formatとstrftimeで2回メソッドを呼ぶより、全部やってくれるメソッドを用意した方がスッキリしそうです

Reply:
`format_last_modified_time`の中で、`strftime`を使うように修正しました。